Sources/StreamChatUI/Components.swift

```diff
@@ -67,27 +67,205 @@ public struct _Components<ExtraData: ExtraDataTypes> {
+
+    /// The view that shows a single reaction.
+    public var reactionItemView: _ChatMessageReactionsView<ExtraData>.ItemView.Type =
+        _ChatMessageReactionsView<ExtraData>.ItemView.self
+
+    /// The view that shows author avatar in `messageContentView`.
+    public var messageAuthorAvatarView: ChatAvatarView.Type = ChatAvatarView.self
+
+    /// The view that shows thread participant avatar in `messageContentView`.
+    public var messageThreadParticipantAvatarView: ChatAvatarView.Type = ChatAvatarView.self
+
+    /// The view that shows error indicator in `messageContentView`.
+    public var messageErrorIndicator: ChatMessageErrorIndicator.Type = ChatMessageErrorIndicator.self
+
+    /// The view that shows message's file attachments.
+    public var fileAttachmentListView: _ChatMessageFileAttachmentListView<ExtraData>
+        .Type = _ChatMessageFileAttachmentListView<ExtraData>.self
+
+    /// The view that shows a single file attachment.
+    public var fileAttachmentView: _ChatMessageFileAttachmentListView<ExtraData>.ItemView.Type =
+        _ChatMessageFileAttachmentListView<ExtraData>.ItemView.self
+
+    /// The view that shows message's image attachments.
+    public var imageGalleryView: _ChatMessageImageGallery<ExtraData>.Type =
+        _ChatMessageImageGallery<ExtraData>.self
+
+    /// The view that shows an overlay with uploading progress for image attachment that is being uploaded.
+    public var imageUploadingOverlay: _ChatMessageImageGallery<ExtraData>.UploadingOverlay.Type =
+        _ChatMessageImageGallery<ExtraData>.UploadingOverlay.self
+
+    /// The view that shows giphy attachment with actions.
+    public var giphyAttachmentView: _ChatMessageInteractiveAttachmentView<ExtraData>.Type =
+        _ChatMessageInteractiveAttachmentView<ExtraData>.self
+
+    /// The button that shows the attachment action.
+    public var giphyActionButton: _ChatMessageInteractiveAttachmentView<ExtraData>.ActionButton.Type =
+        _ChatMessageInteractiveAttachmentView<ExtraData>.ActionButton.self
+
+    /// The view that shows a content for `.giphy` attachment.
+    public var giphyView: _ChatMessageGiphyView<ExtraData>.Type =
+        _ChatMessageGiphyView<ExtraData>.self
+
+    /// The view that shows a badge on `giphyAttachmentView`.
+    public var giphyBadgeView: _ChatMessageGiphyView<ExtraData>.GiphyBadge.Type = _ChatMessageGiphyView<ExtraData>.GiphyBadge.self
+
+    // MARK: - Channel list components
+
+    /// The logic to generate a name for the given channel.
+    public var channelNamer: ChatChannelNamer<ExtraData> = DefaultChatChannelNamer()
+
+    /// The collection view of the channel list.
+    public var channelListCollectionView: UICollectionView.Type = UICollectionView.self
+
+    /// The collection view layout of the channel list.
+    public var channelListLayout: UICollectionViewLayout.Type = ListCollectionViewLayout.self
```

Comment:
Maybe we also don't need these to be in the components struct. If the customer wants to override the channelList, it can override there and change the UICollectionViewLayout and UICollectionView 🤔 The idea, is that really only the main shared components go into this config. Most likely, the channelListLayout won't be shared with other components.

Reply:
The layout is out custom type so it should stay probably. The collection view for the channel list is removed ✅